src/Microsoft.ML.Core/Utilities/Utils.cs

```diff
@@ -776,6 +776,41 @@ public static T[] BuildArray<T>(int length, Func<int, T> func)
             return result;
         }
 
+        /// <summary>
+        /// Given a predicate, over a range of values defined by a limit calculate
+        /// first the values for which that predicate was true, and second an inverse
+        /// map.
+        /// </summary>
+        /// <param name="schema">The input schema where the predicate can check if columns are active.</param>
+        /// <param name="pred">The predicate to test for various value</param>
+        /// <param name="map">An ascending array of values from 0 inclusive
+        /// to <paramref name="schema.Count"/> exclusive, holding all values for which
+        /// <paramref name="pred"/> is true</param>
+        /// <param name="invMap">Forms an inverse mapping of <paramref name="map"/>,
+        /// so that <c><paramref name="invMap"/>[<paramref name="map"/>[i]] == i</c>,
+        /// and for other entries not appearing in <paramref name="map"/>,
+        /// <c><paramref name="invMap"/>[i] == -1</c></param>
+        public static void BuildSubsetMaps(DataViewSchema schema, Func<DataViewSchema.Column, bool> pred, out int[] map, out int[] invMap)
```

Comment:
>Func<DataViewSchema.Column, bool> pred [](start = 66, length = 38)

Hmmm. Is the goal here to enable more elaborate checking I suppose? Totally fine, just interesting.